langchain/src/document_loaders/fs/openai_whisper_audio.ts

```diff
@@ -0,0 +1,55 @@
+import {
+  Configuration,
+  ConfigurationParameters,
+  CreateTranscriptionResponse,
+  OpenAIApi,
+} from "openai";
+import { Document } from "../../document.js";
+import { BaseDocumentLoader } from "../base.js";
+
+const MODEL_NAME = "whisper-1";
+const MAX_FILE_SIZE = 25 * 1024 * 1024; // 25MB
+
+export class OpenAIWhisperAudio extends BaseDocumentLoader {
```

Comment:
Would be nice to have this extend `BufferLoader` as well to keep the `filePathOrBlob` syntax consistent.

I'm trying to do so now and running into some issues with Blob vs. File and MIME types. 

Apparently OpenAI's new major version SDK will fix a lot of these difficulties though - let's wait on integrating that!

Reply:
Yes, there are issues converting blob to File. We could allow to accept filePathOrFile but this won't gully correspond with the filePathOrBlob.

Are you ok if we go with filePathOrFile for now and align with filePathOrBlob after new OpenAI library release?

Reply:
I've talked with the OpenAI SDK owner who said that the beta versions are pretty stable. Let me get back to you later today with an assessment of how long it'd take/whether it'd be worth it to upgrade to one of those beta versions now and we can decide from there?

Reply:
Sounds good. Thank you!

Reply:
Didn't have time today - give me another day please.

Reply:
@jacoblee93 , did you have a chance to clarify a timeline?

Reply:
Sorry - have been working on it in the background and am going to aim for this week!

Reply:
Hey update here - there were a few too many rough edges with v4 to feel comfortable with adding it so have elected to push it off a little farther.